sw/device/rom_exts/rom_ext_manifest_parser.h

```diff
@@ -0,0 +1,239 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_ROM_EXTS_ROM_EXT_MANIFEST_PARSER_H_
+#define OPENTITAN_SW_DEVICE_ROM_EXTS_ROM_EXT_MANIFEST_PARSER_H_
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/rom_exts/manifest.h"
+
+// Header Extern Guard (so header can be used from C and C++)
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+/**
+ * ROM Extension manifest slot type.
+ */
+typedef uintptr_t rom_ext_manifest_slot_t;
+
+/**
+ * ROM Extension manifest slots base addresses.
+ *
+ * These are intended to be used as an input parameter to
+ * `rom_ext_parameters_get`.
+ */
+extern const rom_ext_manifest_slot_t kRomExtManifestSlotA;
+extern const rom_ext_manifest_slot_t kRomExtManifestSlotB;
+
+/**
+ * ROM Extension parameters required for the manifest parsing.
+ */
+typedef struct rom_ext_manifest {
+  /**
+   * Base address of the manifest in memory.
+   */
+  mmio_region_t base_addr;
+  rom_ext_manifest_slot_t slot;
```

Comment:
I was wondering why this was needed, but then I remembered `base_addr` does not contain a pointer during testing. This is good.